lib/kv-extreme-scale.js

```diff
@@ -322,17 +291,21 @@ ExtremeScaleKVConnector.prototype._query = function(query, callback) {
     json: true,
   };
 
-  this.request(requestOptions, (err, res, body) => {
+  this.request(modelName, requestOptions, (err, res, body) => {
     callback(err, body);
   });
 };
 
-ExtremeScaleKVConnector.prototype.request = function(options, callback) {
-  options = extend({
-    baseUrl: this._url,
+ExtremeScaleKVConnector.prototype.request =
+function(modelName, options, callback) {
+  const mapname = this.mapname(modelName);
+  const baseUrl = (options.baseUrl || this._url) + '/' + mapname;
+
+  options = Object.assign(options, {
```

Comment:
`Object.assign` modifies the first argument. As a result, the `options` argument passed to the method is modified too, which I consider evil.

Here is a better solution:

```js
options = Object.assign({}, options, {
  // ...
});
```